terraform/modules/opensearch_domain/opensearch.tf

```diff
@@ -0,0 +1,29 @@
+resource "aws_opensearch_domain" "opensearch" {
+  domain_name    = var.domain_name
+  engine_version = var.engine
+
+  advanced_security_options {
+    internal_user_database_enabled = true
```

Comment:
Let's make a variable for `internal_user_database_enabled` that defaults to true